lib/dirList.js

```diff
@@ -1,10 +1,60 @@
 'use strict'
 
+const os = require('node:os')
 const path = require('node:path')
 const fs = require('node:fs/promises')
-const pLimit = require('p-limit')
+const fastq = require('fastq')
+const fastqConcurrency = os.cpus().length
```

Comment:
I would say that no one want to exhaust all CPU power.
You should always left one core for other task unless only one is available.

```suggestion
const fastqConcurrency = Math.max(1, os.cpus().length - 1)
```

Reply:
I see, it was under the assumption that each CPU has 2 threads, and we'd only be using half

Should I nevertheless decrease it?

Reply:
Yes.

Reply:
`os.cpus()` always return `threads` not the actual core number.

Reply:
> os.cpus() always return threads not the actual core number.

I didn't know that, thanks for letting me know - we could even go for half then, to be honest, because that was my intention

But I'm fine with whatever works the best

Reply:
> I didn't know that, thanks for letting me know - we could even go for half then, to be honest, because that was my intention

It should be fine, since CPU will optimize by it's own task scheduler.
In most case, we are using vCPU (VPS, VM, docker, kubernetes, etc) which is based on CPU threads.